frontend/components/panels/WalletDepositFlowCheckPanel.vue

```diff
@@ -0,0 +1,161 @@
+            needs to be deposited there before the participation can happen.
+        </TextBlock>
+        <Button
+            type="primary"
+            class="w-full mt-2"
+            :disabled="disabled || !isAmountWithinAllowance || !isEnoughInWallet || isEnoughDeposited"
+            :is-loading="isLoading"
+            @click="$emit('deposit')"
+        >
+            Deposit <format-currency :value="minimumDepositAmount" :currency="currency" class="ml-1" />
+        </Button>
+    </BasePanel>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import Button from '~/components/common/BaseButton.vue';
+import BasePanel from '~/components/common/BasePanel.vue';
+import TextBlock from '~/components/common/TextBlock.vue';
+import FormatCurrency from '~/components/utils/FormatCurrency.vue';
+import WalletVatCheckPanel from '~/components/panels/WalletVatCheckPanel.vue';
+import AllowanceAmountCheckPanel from '~/components/panels/AllowanceAmountCheckPanel.vue';
+
+export default Vue.extend({
+    components: {
+        Button,
+        BasePanel,
+        TextBlock,
+        FormatCurrency,
+        WalletVatCheckPanel,
+        AllowanceAmountCheckPanel,
+    },
+    props: {
+        walletAmount: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: undefined,
+        },
+        walletVatAmount: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: undefined,
+        },
+        desiredAmount: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: undefined,
+        },
+        allowanceAmount: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: undefined,
+        },
+        network: {
+            type: String,
+            default: undefined,
+        },
+        tokenAddress: {
+            type: String,
+            default: '',
+        },
+        currency: {
+            type: String,
+            default: 'DAI',
+        },
+        disabled: {
+            type: Boolean,
+            default: false,
+        },
+        isLoading: {
+            type: Boolean,
+            default: false,
+        },
+        isExplanationsShown: {
+            type: Boolean,
+            default: true,
+        },
+    },
+    data: () => ({
+        isAmountWithinAllowance: false,
+        isEnoughInWallet: false,
+    }),
+    computed: {
+        minimumDepositAmount(): BigNumber | undefined {
+            if (!this.desiredAmount || !this.walletVatAmount) {
+                return undefined;
+            }
+            return BigNumber.maximum(0, this.desiredAmount.minus(this.walletVatAmount));
+        },
+        isDesiredAmountValid(): boolean {
+            return this.desiredAmount && !this.desiredAmount.isNaN();
+        },
+        isEnoughDeposited(): boolean {
+            return this.minimumDepositAmount?.isZero() || false;
```

Comment:
In the future, please prefer using [`??` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) when dealing with null produced by `?.`